crates/stc_ts_file_analyzer/src/analyzer/generalize.rs

```diff
@@ -72,7 +72,6 @@ impl Simplifier<'_> {
 impl Fold<Union> for Simplifier<'_> {
     fn fold(&mut self, mut union: Union) -> Union {
         let should_remove_null_and_undefined = union.types.iter().any(|ty| match ty.normalize() {
-            Type::TypeLit(..) => true,
```

Comment:
Currently , I removed only this line. If I can remove `should_remove_null_and_undefined` itself, I will add new commit.

Reply:
I think we should remove all of them

Reply:
Can you remove all of this?

Reply:
I removed!